src/plugins/discover/public/application/view_components/utils/use_search.ts

```diff
@@ -236,6 +241,19 @@ export const useSearch = (services: DiscoverViewServices) => {
           elapsedMs,
         },
       });
+
+      // Add tracking for datasource type and query if the query is not same as last query
+      const query = searchSource.getField('query');
+      const queryStringHistory = data.query.queryString.getQueryHistory();
+      if (
+        query &&
+        !isEqual(query, queryStringHistory.slice(-1)[0]) &&
+        query.dataset?.type &&
+        query?.language
+      ) {
+        trackUiMetric(getDatasourceMetricEventName(query.dataset.type));
```

Comment:
nit: this only tracks on success. did we want to track the ui metric regardless of response?

Reply:
Updated it to track the event before the search actually happens.